### PR TITLE
Spark session client

### PR DESCRIPTION
**Pull request by jcamachor**
Relates to #17.

This patch contains multiple changes to enable testing using Spark session client, and in general, non-JDBC based connections. Concretely:
- Subclassing of ConnectionManager into JDBCConnectionManager and SparkConnectionManager.
- Moving configuration input parsing methods into their own utility class.
- TelemetryRegistry can rely on any ConnectionManager.
- Testing for Delta using session client (Hudi and Iceberg disabled for the time being, see #71).

The PR includes moving to Java11, since Java17 causes some issues with Spark dependency: https://stackoverflow.com/questions/72724816/running-unit-tests-with-spark-3-3-0-on-java-17-fails-with-illegalaccesserror-cl

### CODE REVIEW

**Comment by jcamachor** [comment]
@ashvina , @anjagruenheid , could any of you review? Thanks

File: src/main/java/com/microsoft/lst_bench/Driver.java [hunk 1]
```diff
@@ -114,10 +115,21 @@ public static void main(String[] args) throws Exception {
...
+      Workload workload,
+      ConnectionsConfig connectionsConfig,
+      ExperimentConfig experimentConfig,
+      TelemetryConfig telemetryConfig)
+      throws Exception {
     // Create connections manager
     Map<String, ConnectionManager> idToConnectionManager = new LinkedHashMap<>();
     for (ConnectionConfig connectionConfig : connectionsConfig.getConnections()) {
-      ConnectionManager connectionManager = ConnectionManager.from(connectionConfig);
+      ConnectionManager connectionManager = InputToBench.connectionManager(connectionConfig);
```
Comment (ashvina) [inline src/main/java/com/microsoft/lst_bench/Driver.java:132]: nit: the first time I read the class name `InputToBench`, I did not think it is dealing with config and object instantiation. The class appears to be more of a 'BenchmarkObjectFactory` to me.

**Review by ashvina — APPROVED** [review]
LGTM 👍

File: src/main/java/com/microsoft/lst_bench/input/InputToBench.java [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+import com.microsoft.lst_bench.sql.SQLParser;
+import com.microsoft.lst_bench.util.FileParser;
+import com.microsoft.lst_bench.util.StringUtils;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class InputToBench {
```
Comment (anjagruenheid) [inline src/main/java/com/microsoft/lst_bench/input/InputToBench.java:43]: Can you add a class description?

File: src/test/java/com/microsoft/lst_bench/DriverSparkTest.java [hunk 3]
```diff
@@ -15,64 +15,189 @@
...
+        new HashMap<>(Objects.requireNonNull(experimentConfig.getParameterValues()));
+    parameterValues.compute("external_data_path", (k, value) -> tempDir.toString() + value);
+    parameterValues.compute("data_path", (k, value) -> tempDir.toString() + value);
+    return ImmutableExperimentConfig.builder()
+        .from(experimentConfig)
+        .parameterValues(parameterValues)
+        .build();
+  }
+
+  private void createTempDirs(Path tempDir) {
```
Comment (anjagruenheid) [inline src/test/java/com/microsoft/lst_bench/DriverSparkTest.java:153]: Could we do this with a loop and a list of table names to iterate over?

**Review by anjagruenheid — APPROVED** [review]

**Pull request merged**